### ISSUE TITLE
Setting additionnal PATHS in the module generation

### ISSUE THREAD

**Issue by Billae** [LLNL/spack#21763]
Hello,

I have a project which aims to be a Paraview pugin. I have created a spack recipe for it and a spack environment to build it. Now I want to generate modules associated to this env. I use for that _spack module tcl refresh_ and it works fine.

However, as my project is a paraview plugin, I need to "inform" my paraview installation that this plugin is available by setting the **$PV_PLUGIN_PATH** variable. Ideally, when I load my project module, I want this variable automatically set.

I try to find a way to indicate in the spack recipe of my project that during the module generation, it have to set this variable. I have read the spack doc all the day, and I have not find something which do that. My question is : 

Is it a way to do that or not yet ?

Thanks for reading =)
Eloise

**Comment by alalazo** [comment]
Hi @Billae, you can encode modifications to environment variables needed to run your package in the `package.py` file:
```python
def setup_run_environment(self, env):
    env.prepend_path('PV_PLUGIN_PATH', ...)
```
These modifications should be reflected in the module files that get generated.

**Comment by Billae** [comment]
Thank you,
I did see this in my researches, but actually I haven't understood that this was the point i was looking for =).
Eloïse

**Comment by Billae** [comment]
Hello,

I'm sorry to reopen this, but its remain one problem on setting this additionnal path:

To detect my project as a plugin, Paraview need a full path to spack/opt/.../project/lib( or lib64)/paraview-<version of paraview\>/plugin/project.

As when i install my plugin I did not know which is the paraview version associated, I need to make this path generic.
To this end I try to add in my recipe
```yaml
    def setup_run_environment(self, env):
        env.prepend_path('PV_PLUGIN_PATH', join_path(self.prefix, 'lib*', '**', 'plugins','project'))
```

However, the regular expression usage is not detected when i run paraview which say to me "_searching plugins in   spack/opt/.../project/lib*/**/plugin/project._".
When i try to echo this path it works fine:
```bash
> echo $PV_PLUGIN_PATH
> spack/opt/.../project/lib64/paraview-5.9/plugin/project.
```

In short, I need to evaluate this regular expression during the module generation, and if possible in the _setup\_run\_environment_ method. Is it possible ?

Thanks
Eloïse
